TWLight/resources/views.py

```diff
@@ -42,9 +46,46 @@ def get_queryset(self):
         else:
             return Partner.objects.order_by('company_name')
 
+    def get_context_data(self, **kwargs):
+        context = super(PartnersFilterView, self).get_context_data(**kwargs)
+        
+        if self.request.user.is_staff:
+            partners_list = Partner.even_not_available.order_by('company_name')
+        else:
+            partners_list = Partner.objects.order_by('company_name')
+        
+        # Retrieves the short descriptions of all partners from Meta
+        user_language = self.request.LANGUAGE_CODE
+        languages_on_revision_field_short_desc = {}
+        context['short_description'] = {}
+        
+        for every_partner in partners_list:
```

Comment:
We're firing way too much work for each request here. Also, the code here shouldn't be concerned with revision ids or cache invalidation. That's work to do elsewhere, probably `APIPartnerDescriptions`.